src/Shared/ReuseableStringBuilder.cs

```diff
@@ -309,6 +306,9 @@ internal static void Release(StringBuilder returningBuilder)
                 // So the shared builder will be "replaced".
                 if (returningBuilder.Capacity < MaxBuilderSize)
                 {
+                    // ErrorUtilities.VerifyThrow(handouts.TryRemove(returningBuilder, out dummy), "returned but not loaned");
+                    returningBuilder.Clear(); // This is free: it just sets length to zero internally
```

Comment:
Setting length to 0 can [allocate a `char[]`](https://github.com/dotnet/coreclr/blob/eb1d52cc2e2a32ccbead0dac01c99a073640ee13/src/mscorlib/shared/System/Text/StringBuilder.cs#L534-L540) if the builder has moved passed its initial extent; which is definitely case if its being discarded